packages/connect-node/src/private/web-header-to-node-headers.ts

```diff
@@ -67,3 +77,47 @@ export function webHeaderToNodeHeaders(
   }
   return o;
 }
+
+export function parseGrpcTimeoutFromHeader(value: string): number | null {
+  if (value.length === 0) {
+    return null;
+  }
+
+  const result = value.match(/[a-z]+|[^a-z]+/gi); // splits number from character
```

Comment:
`^(\d{1,8})([HMSmun])$` shifts more work to the reg exp. You can safely call `parseInt` on the first capture group, you're guaranteed to only get valid units, and you don't need to test > 99999999 anymore.

Best to add some test cases for this :)